test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataJsonWriterUtilsTests.cs

```diff
@@ -283,23 +283,20 @@ public void WriteError_InnerErrorWithCollectionAndNulls()
         public void ODataInnerErrorToStringTest()
         {
             ODataInnerError innerError = new ODataInnerError();
-            innerError.Properties.Add("ResourceValue", new ODataResourceValue(){Properties = new ODataProperty[] { new ODataProperty() { Name = "PropertyName", Value = "PropertyValue"}, new ODataProperty(){Name = "NullProperty", Value = new ODataNullValue()}}});
+            innerError.Properties.Add("ResourceValue", new ODataResourceValue() { Properties = new[] { new ODataProperty { Name = "PropertyName", Value = "PropertyValue" }, new ODataProperty { Name = "NullProperty", Value = new ODataNullValue() } } });
             innerError.Properties.Add("NullProperty", new ODataNullValue());
-            innerError.Properties.Add("CollectionValue", new ODataCollectionValue(){Items = new List<object>() {new ODataNullValue(), new ODataPrimitiveValue("CollectionValue"), new ODataPrimitiveValue(1)}});
+            innerError.Properties.Add("CollectionValue", new ODataCollectionValue() { Items = new List<object>() { new ODataNullValue(), new ODataPrimitiveValue("CollectionValue"), new ODataPrimitiveValue(1) } });
 
-            string result = innerError.ToJson();
+            string result = innerError.ToJsonString();
 
-            Assert.Equal("{\"message\":\"\"," +
-                               "\"type\":\"\"," +
-                               "\"stacktrace\":\"\"," +
-                               "\"innererror\":{}," +
-                               "\"ResourceValue\":{" +
-                                    "\"PropertyName\":\"PropertyValue\"," +
-                                    "\"NullProperty\":null" +
-                                    "}," +
-                               "\"NullProperty\":null," +
-                               "\"CollectionValue\":[null,\"CollectionValue\",1]" +
-                               "}", result);
+            Assert.Equal("{\"ResourceValue\":{" +
+                "\"PropertyName\":\"PropertyValue\"," +
+                "\"NullProperty\":null}," +
+                "\"NullProperty\":null," +
+                "\"CollectionValue\":[null,\"CollectionValue\",1]," +
+                "\"message\":\"\"," +
+                "\"type\":\"\"," +
+                "\"stacktrace\":\"\"}", result);

```

Comment:
Is the order meant to change here?

Reply:
I have explained this change in the pull request description:
> When serializing an ODataErrorDetail instance, the OData serializers would write the optional target property ahead of the required message property. I switched the order such that the required message property is written first. This makes logical sense and change of property order does not qualify as a breaking change.

That's why I refactored this plus a few other tests